File: Exam_Portal_UI/src/app/service/question.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuestionService {

  constructor(private http : HttpClient) { }


  public getQuistionOfList(qid :any){
    return this.http.get(`${baseUrl}/question/quiz/all/${qid}`);
  }

  public getQuistionOfQuizForTest(qid :any){
    return this.http.get(`${baseUrl}/question/quiz/${qid}`);
  }

  public getSingleQuestion(qid :any){
    return this.http.get(`${baseUrl}/question/${qid}`);
  }

  public deleteQuestion(qid :any){
    return this.http.delete(`${baseUrl}/question/${qid}`);
  }

  public addQuestion(questionData :any){
    return this.http.post(`${baseUrl}/question/`,questionData);
  }

  public updateQuestion(questionData :any){
    return this.http.put(`${baseUrl}/question/`,questionData);
  }

  public evalQuizQuestion(questionData :any){
    return this.http.post(`${baseUrl}/question/eval-quiz`,questionData);
  }
}

File: Exam_Portal/src/main/java/com/exam/service/QuestionService.java
package com.exam.service;

import java.util.List;
import java.util.Set;

import com.exam.model.exam.Question;
import com.exam.model.exam.Quiz;

public interface QuestionService {

	public Question addQuestion(Question question);

	public Question updateQuestion(Question question);

	public List<Question> getQuestions();

	public Question getQuestion(Long qid);

	public void deleteQuestion(Long qid);
	
	public Set<Question> getQuestionsOfQuiz(Quiz quiz);
	
	public Question getQustionWithAnswer(Long qusId);
}
